lib/vfscore/automount.c

```diff
@@ -67,10 +71,103 @@ struct vfscore_volume {
 	unsigned long flags;
 	/* Mount options */
 	const char *opts;
+#if CONFIG_LIBVFSCORE_FSTAB
+	/* Unikraft Mount options, see vfscore_mount_volume() */
+	char *ukopts;
+#endif /* CONFIG_LIBVFSCORE_FSTAB */
 };
 
+#if CONFIG_LIBVFSCORE_FSTAB
+/* Handle `mkpath` Unikraft Mount Option */
+static int vfscore_volume_process_ukopts_do_mkpath(const char *path)
+{
+	const char *curri;
+	char *pos;
+	int rc;
+
+	UK_ASSERT(path);
+
+	curri = path;
+	pos = strchr(path + 1, '/');
```

Comment:
Is the compiler not warning here? `strchr()` should return `const char *`

Reply:
We also assume here that path is longer than 1 Byte (`\0`).
I think we should add another assertion for this function:
```diff
 UK_ASSERT(path);
+UK_ASSERT(path[0] == '/');
```

Reply:
> Is the compiler not warning here? strchr() should return const char *

We generally have so many warnings that I missed this one.